lib/swim.ml

```diff
@@ -0,0 +1,127 @@
+open Base.Hashtbl
+
+type message =
+  | Ping
+  | Acknowledge
+  | PingRequest of Peer.address
+[@@deriving bin_io]
+
+type config = {
+  protocol_period : int;
+  round_trip_time : int;
+  timeout : int;
+}
+
+type t = {
+  config : config;
+  (* TODO: I am totally not sure about this type *)
+  acknowledges : (int, unit Lwt_condition.t) Poly.t;
```

Comment:
Can you explain what this is exactly?

Reply:
In my mind, I just want to store:
(sequence_number, status) for each Peer